File: paper/src/main/java/com/badbones69/crazycrates/managers/crates/other/quadcrates/SessionManager.java
package com.badbones69.crazycrates.managers.crates.other.quadcrates;

import org.bukkit.entity.Player;

public class SessionManager {

    public boolean inSession(Player player) {
        if (QuadCrateManager.getCrateSessions().isEmpty()) return false;

        for (QuadCrateManager quadCrateManager : QuadCrateManager.getCrateSessions()) {
            if (quadCrateManager.getPlayer() == player) return true;
        }

        return false;
    }

    public QuadCrateManager getSession(Player player) {
        for (QuadCrateManager quadCrateManager : QuadCrateManager.getCrateSessions()) {
            if (quadCrateManager.getPlayer() == player) return quadCrateManager;
        }

        return null;
    }

    public static void endCrates() {
        if (!QuadCrateManager.getCrateSessions().isEmpty()) {
            QuadCrateManager.getCrateSessions().forEach(session -> session.endCrateForce(false));
            QuadCrateManager.getCrateSessions().clear();
        }
    }
}

File: plugin/src/main/java/com/badbones69/crazycrates/v2/utils/quadcrates/sessions/SessionManager.kt
package com.badbones69.crazycrates.v2.utils.quadcrates.sessions

import com.badbones69.crazycrates.v2.crates.sessions.QuadCrateSession
import org.bukkit.entity.Player

object SessionManager {

    fun inSession(player: Player): Boolean {
        QuadCrateSession.getCrateSessions().forEach {
            if (it.player == player) return true
        }
        return false;
    }

    fun getSession(player: Player): QuadCrateSession? {
        QuadCrateSession.getCrateSessions().forEach {
            if (it.player == player) return it
        }
        return null
    }

    fun endAllCrates() {
        QuadCrateSession.getCrateSessions().forEach { session -> session.endCrateForce(false) }
        QuadCrateSession.getCrateSessions().clear()
    }
}